guides/client/js-interop.md

```diff
@@ -354,3 +354,47 @@ Hooks.Chart = {
 ```
 
 *Note*: In case a LiveView pushes events and renders content, `handleEvent` callbacks are invoked after the page is updated. Therefore, if the LiveView redirects at the same time it pushes events, callbacks won't be invoked on the old page's elements. Callbacks would be invoked on the redirected page's newly mounted hook elements.
+
+## Locked DOM elements and loading states
+
+Whenever pushings events to the server, such as a button click or form submission, LiveView locks elements in the DOM and applies CSS based loading states as outlined in the [bindings](bindings.md) guide. The lock consists of a `data-phx-ref` attribute, which prevents any inbound patches from
+altering the contents of an element until that ref is acknowledged by the server. This prevents UI race conditions from happenign on the client and provides UI feedback during the round trip.
+
+Custom JavaScript can integrate with the locking events to implement optimistic UI layers, or apply their own custom loading states. The following `CustomEvent`'s are dispatched to an element whenever an event is pushed to the server:
+
+  `phx:lock` – A lock was applied to the element for any phx-binding event.
+    The event `detail` is an object with:
+
+    `event` – The name of the `phx-` event, for exmaple `<button phx-click="clicked">`
+      would set the event as `"clicked"`.
```

Comment:
Took me some re-reading to realise the `event` value is `clicked` not because of `phx-click`.

This tiny change might save some confusion

```suggestion
    `event` – The name of the `phx-` event, for exmaple `<button phx-click="save">`
      would set the event as `"save"`.
```